server/aesdsocket.c

```diff
@@ -0,0 +1,366 @@
+/*************************************************************************************************
+* Filename : aesdsocket.c
+* Author : Maanas Makam Dileep Kumar
+* Description : Server implementation using Sockets.
+* Date : 02/20/2023
+* Reference : https://beej.us/guide/bgnet/html/ & Lecture Videos
+*************************************************************************************************/
+#include <stdio.h>
+#include <stdlib.h>
+#include <syslog.h>
+#include <fcntl.h>
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <signal.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <netdb.h>
+
+#define RET_ERROR (-1)
+#define SERVER_PORT ("9000")
+#define BACKLOG (10)
+#define BUFFER_SIZE (3)
+
+char *recieve_buffer;
+int sockfd;
+int new_fd;
+int tmp_fd;
+
+// Handler for SIGINT and SIGTERM
+void signal_handler(int signum)
+{
+    if (signum == SIGINT) 
+    {
+        syslog(LOG_DEBUG, "Caught signal SIGINT, exiting!!");
+    }
+    else if( signum == SIGTERM)
+    {
+        syslog(LOG_DEBUG, "Caught signal SIGTERM, exiting!!");
+    }
+
+    int ret = close(sockfd);
+    if(ret == -1)
+    {
+        syslog(LOG_ERR, "Close socket error with errno : %d", errno);
+    }
+
+    if( new_fd != -1)
+    {
+        ret = close(new_fd);
+        if(ret == -1)
+        {
+            syslog(LOG_ERR, "Close new_fd error with errno : %d", errno);
+        }
+    }
+
+    ret = close(tmp_fd);
+    if(ret == -1)
+    {
+        syslog(LOG_ERR, "Close storage file error with errno : %d", errno);
+    }
+
+    ret = unlink("/var/tmp/aesdsocketdata");
+    if(ret == -1)
+    {
+        syslog(LOG_ERR, "Unlink error with errno : %d", errno);
+    }
+    closelog();
+
+    exit(0);
```

Comment:
Is `exit` safe to do in a signal handler?

Reply:
No, I did not think about it. Thanks for pointing it out!